jupyter_server/serverapp.py

```diff
@@ -2152,6 +2163,7 @@ def _cleanup(self):
         self.remove_browser_open_files()
         self.cleanup_kernels()
         self.cleanup_terminals()
+        self.cleanup_extensions()
```

Comment:
I think we should probably call `_cleanup_extensons()` before `_cleanup_kernels()` since some extensions may need to interact with or interrupt kernel ops as part of their termination.